spec-main/chromium-spec.ts

```diff
@@ -364,6 +365,77 @@ describe('web security', () => {
   });
 });
 
+ifdescribe(process.platform === 'linux')('subprocesses', () => {
+  let appProcess: ChildProcess.ChildProcessWithoutNullStreams | undefined;
+  afterEach(() => {
+    if (appProcess && !appProcess.killed) {
+      appProcess.kill();
+      appProcess = undefined;
+    }
+  });
+
+  it('does not propagate GDK_BACKEND', async () => {
+    const appPath = path.join(fixturesPath, 'api', 'gdk-backend-check');
+
+    console.log('gdk_backend before shell open', process.env.GDK_BACKEND);
+
+    ChildProcess.execFileSync(path.join(appPath, 'before-test.sh'), { cwd: appPath, encoding: 'utf8' });
+
+    appProcess = ChildProcess.spawn(process.execPath, [path.join(appPath, 'main.js')], { env: { ...process.env, GDK_BACKEND: '' } });
+
+    let output = '';
+    appProcess.stdout.on('data', (data) => { output += data; });
+    let stderr = '';
+    appProcess.stderr.on('data', (data) => { stderr += data; });
+
+    const [code, signal] = await emittedOnce(appProcess, 'exit');
+    if (code !== 0) {
+      throw new Error(`Process exited with code "${code}" signal "${signal}" output "${output}" stderr "${stderr}"`);
+    }
+
+    await promisifiedTimers.setTimeout(1000);
+
+    let gdkBackend = fs.readFileSync('/tmp/groot-says', 'utf8');
+    gdkBackend = gdkBackend.replace(/(\r\n|\n|\r)/gm, '');
+
+    console.log('gdkBackend ', gdkBackend);
+    expect(gdkBackend).to.be.empty();
+
+    ChildProcess.execFileSync(path.join(appPath, 'after-test.sh'), { cwd: appPath, encoding: 'utf8' });
+  });
+
+  it('successfully honors GDK_BACKEND set in the subproc', async () => {
+    const appPath = path.join(fixturesPath, 'api', 'gdk-backend-check');
+
+    console.log('gdk_backend before shell open', process.env.GDK_BACKEND);
+
+    ChildProcess.execFileSync(path.join(appPath, 'before-test.sh'), { cwd: appPath, encoding: 'utf8' });
+
+    appProcess = ChildProcess.spawn(process.execPath, [path.join(appPath, 'main.js')], { env: { ...process.env, GDK_BACKEND: 'groot' } });
+
+    let output = '';
+    appProcess.stdout.on('data', (data) => { output += data; });
+    let stderr = '';
+    appProcess.stderr.on('data', (data) => { stderr += data; });
+
+    const [code, signal] = await emittedOnce(appProcess, 'exit');
+    if (code !== 0) {
+      throw new Error(`Process exited with code "${code}" signal "${signal}" output "${output}" stderr "${stderr}"`);
+    }
+
+    await promisifiedTimers.setTimeout(1000);
+
+    let gdkBackend = fs.readFileSync('/tmp/groot-says', 'utf8');
+    gdkBackend = gdkBackend.replace(/(\r\n|\n|\r)/gm, '');
```

Comment:
I'm not 100% sure, but I think `.trim()` will do exactly this

Reply:
@jviotti I see that now. Used the snippet from Shelley's PR